generator/uscore/uscore_generator.rb

```diff
@@ -349,60 +349,73 @@ def create_must_support_test(sequence)
-                  must_support_confirmed[id] = true if resource.extension.any? { |extension| extension.url == url }
-                  break if must_support_confirmed[id]
-                end
-                skip_notification = "Could not find \#{id} in any of the \#{@#{sequence[:resource].underscore}_ary.length} provided #{sequence[:resource]} resource(s)"
-                skip skip_notification unless must_support_confirmed[id]
+            must_support_extensions = {
+              #{extensions_list.join(",\n          ")}
+            }
+            missing_must_support_extensions = must_support_extensions.reject do |_id, url|
+              @#{sequence[:resource].underscore}_ary&.any? do |resource|
+                resource.extension.any? { |extension| extension.url == url }
               end
-      )
-        end
-        elements_list = []
-        sequence[:must_supports].select { |must_support| must_support[:type] == 'element' }.each do |element|
-          element[:path] = element[:path].gsub('.class', '.local_class') # class is mapped to local_class in fhir_models
-          elements_list << "'#{element[:path]}'"
+            end
+          )
         end
 
-        if elements_list.any?
+        if must_support_elements.present?
+          elements_list = must_support_elements.map { |element| "'#{element[:path]}'" }
+
           test[:test_code] += %(
-              must_support_elements = [
-                #{elements_list.join(",\n          ")}
-              ]
-              must_support_elements.each do |path|
-                @#{sequence[:resource].underscore}_ary&.each do |resource|
-                  truncated_path = path.gsub('#{sequence[:resource]}.', '')
-                  must_support_confirmed[path] = true if resolve_element_from_path(resource, truncated_path).present?
-                  break if must_support_confirmed[path]
-                end
-                resource_count = @#{sequence[:resource].underscore}_ary.length
+            must_support_elements = [
+              #{elements_list.join(",\n          ")}
+            ]
+
+            missing_must_support_elements = must_support_elements.reject do |path|
+              truncated_path = path.gsub('#{sequence[:resource]}.', '')
+              @#{sequence[:resource].underscore}_ary&.any? do |resource|
+                resolve_element_from_path(resource, truncated_path).present?
+              end
+            end
+          )
 
-                skip "Could not find \#{path} in any of the \#{resource_count} provided #{sequence[:resource]} resource(s)" unless must_support_confirmed[path]
-              end)
+          if must_support_extensions.present?
+            test[:test_code] += %(
+              missing_must_support_elements += missing_must_support_extensions.keys
+            )
+          end
+
+          test[:test_code] += %(
+            assert missing_must_support_elements.empty?,
```

Comment:
This was a skip before, but this seems like a failure rather than a skip to me.

Reply:
Since there isn't a requirement anywhere that states in a given set of resources you must have all must support elements represented, I think we should keep it a skip.  We've defined skip here to mean "you haven't done anything wrong, you just haven't shown us enough to let us pass you, please choose another patient".

In a practical sense, we let users run the Data Access tests multiple times with different patient ids and compose together a single passing set of tests, but we will only re-run "skipped" tests in this case.  So by failing them on this we won't rerun the sequence if they choose a different patient.

Perhaps this feature is too confusing, and we should just require a single patient has everything, or allow the user to pass in a list of patient ids (comma delimited).